sonaranalyzer-dotnet/src/SonarAnalyzer.CSharp/Rules/LiteralSuffixUpperCase.cs

```diff
@@ -48,14 +49,17 @@ protected override void Initialize(SonarAnalysisContext context)
                     var literal = (LiteralExpressionSyntax)c.Node;
                     var text = literal.Token.Text;
 
-                    if (text[text.Length - 1] == 'l' &&
-                        c.SemanticModel.GetTypeInfo(literal).Type.Is(KnownType.System_Int64))
+                    if (text[text.Length - 1] == 'l' && NotUnsigned(text))
                     {
                         c.ReportDiagnosticWhenActive(Diagnostic.Create(rule, Location.Create(literal.SyntaxTree,
                             new TextSpan(literal.Span.End - 1, 1))));
                     }
                 },
                 SyntaxKind.NumericLiteralExpression);
         }
+
+        // a number may be `2ul` or `2Ul` and it's easier to read
+        private static bool NotUnsigned(string text) =>
```

Comment:
Nice! Since we subscribe to `NumericLiteralExpression` there is no need for `SemanticModel` calls to get the type :) 

Can we use `text.Length >= 3`. I think that if it's a number it can't start with a char so the condition can't be true for length = 2.